nucliadb/nucliadb/ingest/maindb/driver.py

```diff
@@ -85,3 +86,16 @@ async def keys(
         self, match: str, count: int = DEFAULT_SCAN_LIMIT, include_start: bool = True
     ) -> AsyncGenerator[str, None]:
         raise NotImplementedError()
+
+    @asynccontextmanager
+    async def managed_transaction(self) -> AsyncGenerator[Transaction, None]:
```

Comment:
As an idea for renaming, maybe `transaction` is good enough. The clients will write code like this:
```
with driver.transaction() as txn:
    ...
```